compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/SyntaxKind.java

```diff
@@ -201,23 +201,43 @@
     PARAMETER_DOC_REFERENCE_TOKEN(907, "parameter"),
     CONST_DOC_REFERENCE_TOKEN(908, "const"),
 
+    //Literal tokens
     IDENTIFIER_TOKEN(1000),
-    STRING_LITERAL(1001),
-    DECIMAL_INTEGER_LITERAL(1002),
-    HEX_INTEGER_LITERAL(1003),
-    DECIMAL_FLOATING_POINT_LITERAL(1004),
-    HEX_FLOATING_POINT_LITERAL(1005),
+    STRING_LITERAL_TOKEN(1001),
+    DECIMAL_INTEGER_LITERAL_TOKEN(1002),
+    HEX_INTEGER_LITERAL_TOKEN(1003),
+    DECIMAL_FLOATING_POINT_LITERAL_TOKEN(1004),
+    HEX_FLOATING_POINT_LITERAL_TOKEN(1005),
     XML_TEXT_CONTENT(1006),
     TEMPLATE_STRING(1007),
 
+    //Documentation
+    DOCUMENTATION_DESCRIPTION(1100),
+    PARAMETER_NAME(1101),
+    BACKTICK_CONTENT(1102),
+    DEPRECATION_LITERAL(1103),
+    DOCUMENTATION_STRING(1104),
+
+    INVALID_TOKEN(1190),
+
+    //-----------------------------------------------non-terminal-kinds-----------------------------------------------
+
+    //Basic literal kinds
+    STRING_LITERAL(1200),
+    NUMERIC_LITERAL(1201),
+    BOOLEAN_LITERAL(1202),
+    NIL_LITERAL(1203),
```

Comment:
Do we need separate kinds for `null` literal and `()` literal?

Reply:
No, since `null` and `()` are both nil literal I have used a single kind.

Reply:
used seprate kinds